proto/finalitygadget.proto

```diff
@@ -0,0 +1,53 @@
+syntax = "proto3";
+
+package proto;
+
+option go_package = "github.com/babylonlabs-io/finality-gadget/proto";
+
+service FinalityGadget {
+  // InsertBlock inserts a new block into the finality gadget db
+  rpc InsertBlock(BlockInfo) returns (InsertBlockResponse);
```

Comment:
does this mean anyone can insert block to the DB? i think we shouldn't expose this as a protobuf msg. "inserting block" should be an internal task

the gadget should only expose "read" RPCs imo